src/flag_gems/ops/diagonal.py

```diff
@@ -0,0 +1,26 @@
+import logging
+
+import torch
+import triton
+
+from ..utils import pointwise_dynamic
+
+
+@pointwise_dynamic(is_tensor=[True], promotion_methods=[(0, "DEFAULT")])
+@triton.jit
+def copy_func(x):
+    return x
+
+
+def backward(grad_output, input_sizes, offset, dim1, dim2):
+    grad_input = torch.zeros(
+        input_sizes, dtype=grad_output.dtype, device=grad_output.device
+    )
+    diag = torch.diagonal(grad_input, offset, dim1, dim2)
+    copy_func.instantiate(grad_output.ndim)(grad_output, out0=diag)
```

Comment:
since torch.zeros also calls for a kernel, there exist two kernels indeed. I wonder if it's feasible to initialize grad_input as an empty tensor, and assign it in one kernel function.

Reply:
Got it, I'll give it a try.